katapult.go

```diff
@@ -92,13 +103,70 @@ func New(opts ...Opt) (*Client, error) {
 	return c, nil
 }
 
-type Client struct {
-	HTTPClient *http.Client
-	Codec      codec.Codec
+func (c *Client) DoIt(
+	ctx context.Context,
+	request *Request,
+	v interface{},
```

Comment:
I want to try and think of a better name than v? It's hard to tell if this is the sent body or the received body.

Reply:
I don't really have an opinion on the name of `v` here. Only reason I chose it is cause it seems to pretty common within Go's stdlib, like `encoding/json`'s [`Marshal`](https://pkg.go.dev/encoding/json#Marshal)/[`Unmarshal`](https://pkg.go.dev/encoding/json#Unmarshal) for example.

Also I can't really think of a properly fitting name... lol